File: frontend/src/domain/transcription/value-objects/language-code.ts
/**
 * Language code value object.
 */

export class LanguageCode {
    private constructor(public readonly code: string) {
        if (code.length !== 2) {
            throw new Error('Language code must be 2 characters (ISO 639-1)');
        }
        if (!/^[a-z]{2}$/i.test(code)) {
            throw new Error('Language code must contain only letters');
        }
    }

    static create(code: string): LanguageCode {
        return new LanguageCode(code.toLowerCase());
    }

    static english(): LanguageCode {
        return new LanguageCode('en');
    }

    static spanish(): LanguageCode {
        return new LanguageCode('es');
    }

    static french(): LanguageCode {
        return new LanguageCode('fr');
    }

    static german(): LanguageCode {
        return new LanguageCode('de');
    }

    equals(other: LanguageCode): boolean {
        return this.code === other.code;
    }
}

File: backend/streamcraft/domain/transcription/value_objects/language_code.py
"""Language code value object."""

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class LanguageCode:
    """ISO 639-1 language code."""

    code: str

    def __post_init__(self) -> None:
        """Validate language code."""
        if len(self.code) != 2:
            raise ValueError("Language code must be 2 characters (ISO 639-1)")
        if not self.code.isalpha():
            raise ValueError("Language code must contain only letters")

    @classmethod
    def english(cls) -> "LanguageCode":
        """Create English language code."""
        return cls(code="en")

    @classmethod
    def spanish(cls) -> "LanguageCode":
        """Create Spanish language code."""
        return cls(code="es")

    @classmethod
    def french(cls) -> "LanguageCode":
        """Create French language code."""
        return cls(code="fr")

    @classmethod
    def german(cls) -> "LanguageCode":
        """Create German language code."""
        return cls(code="de")
